baselines/fedavgm/README.md

```diff
@@ -0,0 +1,230 @@
+
+| Dataset | # classes | # partitions | partition method | partition settings|
+| :------ | :---: | :---: | :---: | :---: |
+| CIFAR-10 | 10 | `num_clients` | Latent Dirichlet Allocation (LDA) | `concentration` |
+| Fashion-MNIST | 10 | `num_clients` | Latent Dirichlet Allocation (LDA) | `concentration` |
+
+**Data distribution:** The following figure illustrates the use of multiple `concentration` values to generate the data distribution over 30 clients for CIFAR-10 (10 classes) - [source code](fedavgm/utils.py):
+
+![](_static/concentration_cifar10_v2.png)
+
+**Training Hyperparameters:**
+The following table shows the main hyperparameters for this baseline with their default value (i.e. the value used if you run `python main.py` directly)
+
+| Description | Default Value |
+| ----------- | ----- |
+| total clients | 10 |
+| number of rounds | 5 |
+| model | CNN |
+| strategy | Custom FedAvgM |
+| dataset | CIFAR-10 |
+| concentration | 0.1 |
+| fraction evaluate | 0 |
+| num cpus | 1 |
+| num gpus | 0 |
+| server momentum | 0.9 |
+| server learning rate | 1.0 |
+| server reporting fraction | 0.05 |
+| client local epochs | 1 |
+| client batch size | 64 |
+| client learning rate | 0.01 |
+
+### Custom FedAvgM
+In contrast with the original implementation from Flower v1.5.0, in this baseline we introduce the Nesterov accelerated gradient as part of the momentum applied in the server model. To use the original Flower implementation, use the argument `strategy=fedavgm`. By default, the custom implementation is used. But, you can also refer to it on the command line as `strategy=custom-fedavgm`.
+
+## Specifying the Python Version
+This baseline was tested with Python 3.10.6.
+
+By default, Poetry will use the Python version in your system. In some settings, you might want to specify a particular version of Python to use inside your Poetry environment. You can do so with [`pyenv`](https://github.com/pyenv/pyenv). Check the documentation for the different ways of installing `pyenv`, but one easy way is using the [automatic installer](https://github.com/pyenv/pyenv-installer):
+
+```bash
+curl https://pyenv.run | bash # then, don't forget links to your .bashrc/.zshrc
+```
+
+You can then install any Python version with `pyenv install <python-version>` (e.g. `pyenv install 3.10.6`). Then, in order to use that version for this baseline, you'd do the following:
+
+```bash
+# cd to your baseline directory (i.e. where the `pyproject.toml` is)
+pyenv local 3.10.6
+
+# set that version for poetry
+poetry env use 3.10.6
+
+# then you can install your Poetry environment (see the next setp)
+```
```

Comment:
Could we make this a sub-section inside "Environment Setup"? (i.e. just adding another `#`)

Reply:
Moved the Python version to be under the "Environment Setup" section.